overlord/patch/patch.go

```diff
@@ -149,7 +149,7 @@ func maybeResetSublevelForLevel60(s *state.State, sublevel *int) error {
 	var sublevelResetTime time.Time
 	lastRefresh, err := getCoreRefreshTime()
 	if err != nil {
-		return fmt.Errorf("cannot determine core refresh time: %s", err)
+		logger.Noticef("WARNING: cannot determine core refresh time: %s", err)
```

Comment:
I think it's fine. I'd add a comment here, something like:
```Edge case (an error) where current symlink for core is missing. In such case we don't error out as this aborts snapd startup, but instead carry on to give snapd chance to recover (e.g. refresh core). Note, because lastRefresh time cannot be determined here, we reset sublevel back to 0 and also sets patch-sublevel-reset timestamp, which means we will run sublevel patches twice (now, in current iteration of patches and then after next core refresh).```

This is fine because sublevel patches were intended to be idempotent, and the aim of this logic and remebering patch-sublevel-reset was simply optimization (avoiding re-running the patch when not needed anymore).

Reply:
this doesn't look right, if for some reason we restart again too soon during the next refresh that add patches it seems we wouldn't run the subpatches, we should probably set lastRefresh to now in this case perhaps?

Reply:
actually the safest would seem to unset the attribute is  lastRefresh.IsZero